### PR TITLE
CPU Plugin refactoring: class names

### PR DESCRIPTION
**Pull request by vladislav-volkov**
### Details:
 - CPU Plugin refactoring: class names and namespaces renaming

### Tickets:
 - CVS-74811

### CODE REVIEW

**Review by tgubanova-lohika — CHANGES_REQUESTED** [review]
Stop russian aggression in Ukraine!

**Comment by dmitry-gorokhov** [comment]
@maxnick could you please review?

File: src/plugins/intel_cpu/src/utils/shape_inference/static_shape.hpp [hunk 1]
```diff
@@ -17,7 +17,9 @@
 namespace ov {
 namespace op {
     struct AutoBroadcastSpec;
-}
+}   // namespace op
+
+namespace intel_cpu {
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/utils/shape_inference/static_shape.hpp:22]: I'm not sure we need to put the shape inference related stuff to the `intel_cpu` namespace. Even though this subsystem is formally housed in the CPU plugin, as far as I know it is meant to be used by all the other plugins as well.

File: docs/IE_PLUGIN_DG/ExecutableNetwork.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ The implementation `CompileNetwork` is fully device-specific.
 The function accepts a const shared pointer to `ngraph::Function` object and performs the following steps:
 
 1. Applies ngraph passes using `TransformNetwork` function, which defines plugin-specific conversion pipeline. To support low precision inference, the pipeline can include Low Precision Transformations. These transformations are usually hardware specific. You can find how to use and configure Low Precisions Transformations in [Low Precision Transformations](@ref openvino_docs_IE_DG_lpt) guide.
-2. Maps the transformed graph to a backend specific graph representation (for example, to MKLDNN graph for Intel CPU).
+2. Maps the transformed graph to a backend specific graph representation (for example, to oneDNN graph for Intel CPU).
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply docs/IE_PLUGIN_DG/ExecutableNetwork.md:41]: fixed

File: src/plugins/intel_cpu/src/infer_request.cpp [hunk 3]
```diff
@@ -790,7 +793,7 @@ InferenceEngine::Blob::Ptr ov::intel_cpu::MKLDNNInferRequest::GetBlob(const std:
                     externalPtr[name] = _inputs[name]->buffer();
                 }
             } else {
-                IE_THROW() << "Blob with name: " << name << " exists in MKLDNN graph, but absents in network inputs";
+                IE_THROW() << "Blob with name: " << name << " exists in graph, but absents in network inputs";
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/infer_request.cpp:796]: fixed

File: src/plugins/intel_cpu/src/nodes/conv.cpp [hunk 4]
```diff
@@ -511,17 +513,17 @@ void MKLDNNConvolutionNode::setPostOps(mkldnn::primitive_attr &attr, const Vecto
...
+        if (auto* eltwiseNode = dynamic_cast<Eltwise *>(node.get())) {
             if (eltwiseNode->isSpecialConvolutionAddFusing()) {
                 if (withSumBroadcast) {
                     break;
                 }
-                ops.append_sum(1.0, MKLDNNExtensionUtils::IEPrecisionToDataType(eltwisePrecision));
+                ops.append_sum(1.0, ExtensionUtils::IEPrecisionToDataType(eltwisePrecision));
             } else {
-                if (useLegacyPostOps || eltwiseNode->getMKLDNNAlgorithm() != mkldnn::algorithm::undef) {
+                if (useLegacyPostOps || eltwiseNode->getOneDNNAlgorithm() != mkldnn::algorithm::undef) {
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/nodes/conv.cpp:526]: DNNL transition will be done in a separate PR

File: src/plugins/intel_cpu/src/nodes/eltwise.cpp [hunk 5]
```diff
@@ -2023,40 +2070,49 @@ bool MKLDNNEltwiseNode::canBeInPlace() const {
...
 }
 
 template <typename T>
-void MKLDNNEltwiseNode::appendPostOpsImpl(mkldnn::post_ops& ops, const VectorDims &postOpDims, std::vector<T>& postOpsMem) {
+void Eltwise::appendPostOpsImpl(mkldnn::post_ops& ops, const VectorDims &postOpDims, std::vector<T>& postOpsMem) {
     const std::string errorPrefix = "Appending Eltwise node with name '" + getName() + "' ";
 
-    if (getMKLDNNAlgorithm() != mkldnn::algorithm::undef) {
-        switch (getMKLDNNAlgorithm()) {
+    if (getOneDNNAlgorithm() != mkldnn::algorithm::undef) {
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/nodes/eltwise.cpp:2114]: fixed

File: src/plugins/intel_cpu/src/descriptor.h [hunk 6]
```diff
@@ -8,48 +8,51 @@
 #include <string>
 #include "mkldnn/ie_mkldnn.h"
 
-class MKLDNNDescriptor {
+namespace ov {
+namespace intel_cpu {
+
+class Descriptor {
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/descriptor.h:14]: fixed

File: src/plugins/intel_cpu/src/extension_utils.h [hunk 7]
```diff
@@ -18,7 +18,7 @@ namespace intel_cpu {
 
 class DnnlMemoryDesc;
 
-class MKLDNNExtensionUtils {
+class ExtensionUtils {
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/extension_utils.h:21]: Renamed to DnnlExtensionUtils

File: src/plugins/intel_cpu/src/utils/shape_inference/static_shape.hpp (hunk 1 above)
Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/utils/shape_inference/static_shape.hpp:22]: All CPU plugin related code should be inside intel cpu namespace. If it is planned to use it in other plugins, it should be moved to a new place, for instance to transformation library.

File: src/plugins/intel_cpu/src/nodes/generic.cpp [hunk 8]
```diff
@@ -81,67 +84,42 @@ void MKLDNNGenericNode::initSupportedPrimitiveDescriptors() {
...
-            }
-
-            for (const auto& impl : impls_no_exec) {
-                if (auto exec_impl = std::dynamic_pointer_cast<InferenceEngine::ILayerExecImpl>(impl)) {
-                    impls.emplace_back(exec_impl);
-                }
-            }
-        }
+        if (impls.empty())
+            return false;
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/nodes/generic.cpp:111]: It is obsolete and not used. Now the CPU plugin uses a different approach for extensions. See extension.h.

File: src/plugins/intel_cpu/src/cpu_types.h [hunk 9]
```diff
@@ -15,7 +15,7 @@ namespace intel_cpu {
 using Dim = std::size_t;
 using VectorDims = std::vector<Dim>;
 
-enum Type {
+enum class Type {
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/cpu_types.h:18]: Without a "class enum", all enum members are visible in the intel_cpu namespace and conflict with class names.

File: src/common/legacy/src/transformations/convert_opset1_to_legacy/convert_strided_slice_to_crop.cpp [hunk 10]
```diff
@@ -204,7 +204,7 @@ ngraph::pass::ConvertStridedSliceToCropMatcher::ConvertStridedSliceToCropMatcher
         }
 
         auto data_node_shape = data_output.get_shape();
-        // MKLDNN: "Crop supports only 2d, 4d and 5d blobs."
+        // INTEL_CPU: "Crop supports only 2d, 4d and 5d blobs."
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply src/common/legacy/src/transformations/convert_opset1_to_legacy/convert_strided_slice_to_crop.cpp:205]: INTEL_CPU removed

File: src/plugins/intel_cpu/src/utils/shape_inference/static_shape.hpp (hunk 1 above)
Comment (maxnick) [reply src/plugins/intel_cpu/src/utils/shape_inference/static_shape.hpp:22]: @dmitry-gorokhov , @jane-intel,  could you please express your opinion on this matter?

Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/utils/shape_inference/static_shape.hpp:22]: For me it is ok to place shape inference under cpu plugin macros. It doesn't bring any overhead its future extraction.

**Review by maxnick — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**